DXRando/DeusEx/Classes/DXRLamp2.uc

```diff
@@ -1,17 +1,60 @@
 //=============================================================================
-// Lamp2 except Warm White.  Fuck daylight bulbs.
+// Lamp2 except with a warm white light at the top.
+// Fuck daylight bulbs.
 // Localization files overwrite the ItemName,
 // so really force it to not be halogen
 //=============================================================================
 class DXRLamp2 injects #var(prefix)Lamp2;
 
+var DynamicLight lt;
+
+function InitLight()
+{
+    local vector loc;
+
+    if (lt == None) {
+        lt = Spawn(class'DynamicLight');
+
+        loc = Location;
+        loc.z += CollisionHeight * 1.35;
+        lt.SetLocation(loc);
```

Comment:
```suggestion
        lt.SetLocation(loc);
        lt.SetBase(self);
```

Reply:
did you test destroying the lamp? because there's nothing here that destroys the DynamicLight